.eslintrc.js

```diff
@@ -7,17 +7,32 @@ module.exports = {
         es2018: true,
         jest: true,
     },
-    plugins: ['prettier'],
+    extends: [
+        "airbnb-base",
+        "airbnb-typescript/base"
+    ],
+    plugins: ['prettier', '@typescript-eslint'],
     extends: ['plugin:prettier/recommended'],
     globals: {
         process: 'readonly',
     },
     rules: {
         'prettier/prettier': 'warn',
         'prefer-const': ['warn'],
-        'no-console': ['warn', { allow: ['warn', 'error', 'info'] }],
+        'no-console': ['error', { allow: ['warn', 'error', 'info'] }],
         curly: 'warn',
         'no-undef': 'off',
         'eol-last': ['error', 'always'],
+        'max-len': ['error', { 'code': 120 }],
+        'no-return-await': [
+            'error'
+        ],
+        'import/prefer-default-export': 'off',
     },
+    ignorePatterns: ['rules/**/*.js', 'test/**/*.js', 'index.js', 'bin/**', '.eslintrc.js'],
```

Comment:
Why do you need to ignore all of those? If after closing this PR we still expect some js files to be present in the repo (which is what I expect, since we only need to write rules in typescript), then I would not exclude all other js files that will stay in the project